app/api/v1/users.py

```
@@ -0,0 +1,69 @@
+    try:
+        await memory_service.add_user(user_id)
+        return {"user_id": user_id}
+    except Exception as e:
+        logger.error(f"Failed to add user: {e}")
+        raise HTTPException(status_code=500, detail="Failed to add user")
+
+
+@router.delete("/{user_id}", response_model=UserResponse)
+async def delete_user(
+    user_id: str,
+    memory_service: MemoryService = Depends(get_memory_service),
```

Ruff (0.8.2)

Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)